src/workerd/jsg/modules.h

```diff
@@ -650,6 +635,19 @@ v8::MaybeLocal<v8::Promise> dynamicImportCallback(v8::Local<v8::Context> context
     return v8::Local<v8::Promise>();
   };
 
+  // The specification for import attributes strongly recommends that embedders
+  // reject import attributes and types they do not understand/implement. This
+  // is because import attributes can alter the interpretation of a module and
+  // are considered to be part of the unique key for caching a module.
+  // Throwing an error for things we do not understand is the safest thing to do.
+  // However, historically we have not followed this guideline in the spec
+  // and unfortunately there are applications deployed that will break if we
+  // started enforcing that guideline without a compat flag.
+  if (!import_attributes.IsEmpty() && import_attributes->Length() > 0 &&
+      js.getThrowOnUnrecognizedImportAssertion()) {
+    return makeRejected(js.v8Error("Unrecognized import attributes specified"));
+  }
```

Comment:
Not a request, but a recommendation, feel free to ignore - it might actually be advisable here to provide a more specific message for the `type` attribute, where the `type` attribute is recognized, just not supported. So we could have the error be "Import attribute type 'css' is not currently supported" say, at least allowing a more specific cause to the user. Further it could help to note the module which has the error in the error message in case it was within a dependency. This would help dependency errors on import attributes to be actionable fixes at least.

Reply:
I think the main use case here is running code that uses `import json from './file.json' with { type: 'json' }` and getting an error that doesn't help guide the user to the correction.